test/integration/image_test.go

```diff
@@ -0,0 +1,127 @@
+//go:build integration
+
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"io"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+// TestImageBuild makes sure the 'minikube image build' command works fine
+func TestImageBuild(t *testing.T) {
+	type validateFunc func(context.Context, *testing.T, string)
+	profile := UniqueProfileName("image")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(15))
+	defer Cleanup(t, profile, cancel)
+
+	// Serial tests
+	t.Run("serial", func(t *testing.T) {
+		tests := []struct {
+			name      string
+			validator validateFunc
+		}{
+			{"NormalBuild", validateNormalImageBuild},
+			{"BuildWithBuildArg", validateImageBuildWithBuildArg},
+			{"BuildWithDockerIgnore", validateImageBuildWithDockerIgnore},
+			{"BuildWithSpecifiedDockefile", validateNormalImageBuildWithSpecifiedDockerfile},
+			{"validateImageBuildWithBuildEnv", validateImageBuildWithBuildEnv},
+		}
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				tc.validator(ctx, t, profile)
+				if t.Failed() && *postMortemLogs {
+					PostMortemLogs(t, profile)
+				}
+			})
+		}
+	})
+}
+
+// validateNormalImageBuild is normal test case for minikube image build, with -t parameter
+func validateNormalImageBuild(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := []string{"image", "build", "-t", "aaa:latest", "./testdata/image-build/test-normal", "-p", profile}
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Fatalf("failed to build image with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// validateNormalImageBuildWithSpecifiedDockerfile is normal test case for minikube image build, with -t and -f parameter
+func validateNormalImageBuildWithSpecifiedDockerfile(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := []string{"image", "build", "-t", "aaa:latest", "-f", "inner/Dockerfile", "./testdata/image-build/test-f", "-p", profile}
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Fatalf("failed to build image with args: %q : %v", rr.Command(), err)
+	}
+	_, err = io.ReadAll(rr.Stdout)
```

Comment:
I don't really see what this is checking, we're throwing away that value we're reading. Plus we could simply get this value with `rr.Stdout.String()`. I think this should be removed.